quinn-proto/src/connection.rs

```diff
@@ -2605,41 +2553,21 @@ where
         }
 
         // STREAM
-        while buf.len() + frame::Stream::SIZE_BOUND < max_size {
-            let mut stream = match space.pending.stream.pop_front() {
-                Some(x) => x,
-                None => break,
-            };
-            if self
+        while buf.len() + frame::Stream::SIZE_BOUND < max_size && space_id == SpaceId::Data {
```

Comment:
The restriction to `SpaceId::Data` is new, but seems mostly unrelated to this change?

Reply:
This requirement has always existed, but in the past it was accomplished by never writing to `pending.stream` for spaces other than `Data`. With this change, stream buffers are (rightly) no longer isolated per-space, so we need to be more explicit, or else stream frames would illegally arise e.g. in `Initial` packets.